userguide/analytics/userguide_analytics.adoc

```diff
@@ -1,63 +1,595 @@
+|sourceTableName   |Name of the source table/view. 
+|sourceName   |Name of the remote database   
+|sourceQuery   |SQL query to be executed. 
+|refreshProcedureName   |Name of the stored procedure    
+|refreshFrequency   |Frequency with which the refresh procedure will be executed. Possible values are `HOURLY` and `DAILY`      
+|refreshHourOfDayGmt   |Hour of the day when the procedure will be executed    
+|===
+
+
+=== Other Report Operations
+
+Once a report is created, you can perform several operations on it as explained below.
+
+==== Retrieve Report Configuration
+
+You can retrieve a report configuration by specifying the report name:
+
+[source,bash]
+----
+curl -u admin:password \
+     -H "X-Killbill-ApiKey:bob" \
+     -H "X-Killbill-ApiSecret:lazar" \
+     "http://127.0.0.1:8080/plugins/killbill-analytics/reports/new_accounts_per_day"
+----
+
+This returns the report configuration (`reportName`, `reportPrettyName`, etc. fields) for the `new_accounts_per_day` report.
+
+==== Update a Report
+
+You can update a report (edit all fields except `reportName`):
+
+[source,bash]
+----
+curl -v \
+     -X PUT \
+     -u admin:password \
+     -H "X-Killbill-ApiKey:bob" \
+     -H "X-Killbill-ApiSecret:lazar" \
+     -H 'Content-Type: application/json' \
+     -d '{"reportPrettyName": "New accounts created (refreshed at 2am GMT)",
+          "refreshFrequency": "DAILY",
+          "refreshHourOfDayGmt": 2}' \
+     "http://127.0.0.1:8080/plugins/killbill-analytics/reports/new_accounts_per_day"
+----
+
+This updates the `reportPrettyName`, `refreshFrequency` and `refreshHourOfDayGmt` fields for the `new_accounts_per_day` report.
+
+==== Delete a Report
+
+You can delete a report by specifying the report name:
+
+[source,bash]
+----
+curl -v \
+     -X DELETE \
+     -u admin:password \
+     -H "X-Killbill-ApiKey:bob" \
+     -H "X-Killbill-ApiSecret:lazar" \
+     "http://127.0.0.1:8080/plugins/killbill-analytics/reports/new_accounts_per_day"
+----
+
+This deletes the `new_accounts_per_day` report.
+
+==== Download Report Data
+
+You can download the data behind your report as `csv` or `json`:
+
+[source,bash]
+----
+curl -u admin:password \
+     -H "X-Killbill-ApiKey:bob" \
+     -H "X-Killbill-ApiSecret:lazar" \
+     'http://127.0.0.1:8080/plugins/killbill-analytics/reports?name=new_accounts_per_day&format=csv'
+----
+
+This returns the data behind the `new_accounts_per_day` report in `csv` format.
+
+==== Retrieve Report SQL Query
+
+You can retrieve the SQL query corresponding to your report:
+
+[source,bash]
+----
+curl -v \
+     -u admin:password \
+     -H "X-Killbill-ApiKey:bob" \
+     -H "X-Killbill-ApiSecret:lazar" \
+     "http://127.0.0.1:8080/plugins/killbill-analytics/reports?name=report_accounts_summary&startDate=2018-01-01&endDate=2022-01-01&sqlOnly=true"
+----
+
+This returns the SQL query corresponding to the `report_accounts_summary` report.
+
+==== Refresh a report - to be confirmed
```

Comment:
```suggestion
==== Refresh a Report - to be confirmed
```

Reply:
Fixed, added an explanation too. @pierre, I see that this endpoint requires a request body to be specified along with the `shouldRefresh=true`. Without a request body, the endpoint results in an exception. Shouldn't the endpoint work with just the `shouldRefresh=true` parameter? If you agree, maybe this is something we need to investigate further, I will make a note of this.

Reply:
If `shouldRefresh=true`, the endpoint is used to refresh the report. Otherwise, the report definition is updated (and a body is required): https://github.com/killbill/killbill-analytics-plugin/blob/0dbef7add941547d6d49371b48ab45ce54084e25/src/main/java/org/killbill/billing/plugin/analytics/http/ReportsResource.java#L259-L263

Reply:
If `shouldRefresh=true`, the endpoint is used to refresh the report. Otherwise, the report definition is updated (and a body is required): https://github.com/killbill/killbill-analytics-plugin/blob/0dbef7add941547d6d49371b48ab45ce54084e25/src/main/java/org/killbill/billing/plugin/analytics/http/ReportsResource.java#L259-L263

Reply:
Thanks. From what I see, a request body does need to be specified when `shouldRefresh=true` is true. I will make a note to investigate this further.

Reply:
👍